### PR TITLE
Fixed TS tests with JS contract (close #581)

### LINKED ISSUE #581: TypeScript tests fail to run with a JavaScript smart contract
TypeScript tests fail to run for a generated JavaScript smart contract.
This is because there is no `tsconfig.json` created on the project or test generation.
Additionally, the `typescript` and `ts-node` dependencies are not included in a JavaScript smart contract, so this will cause TypeScript tests to fail.

When generating TypeScript tests, we should consider installing these two dependencies and creating a `tsconfig.json` file.

### PR DESCRIPTION
**Pull request by Jakeeyturner**
Signed-off-by: Jake Turner <jaketurner25@live.com>

### CODE REVIEW

File: client/src/commands/testSmartContractCommand.ts [hunk 1]
```diff
@@ -223,13 +223,42 @@ export async function testSmartContract(chaincode?: InstantiatedContractTreeItem
...
+                './functionalTests/**/*'
+            ],
+            exclude: [
+                'node_modules'
+            ]
+        };
+
+        // Try to create tsconfig.json file
+        try {
+            await fs.writeJson(tsConfigPath, tsConfigContents, {spaces: '\t'});
```
Comment (caroline-church) [inline client/src/commands/testSmartContractCommand.ts]: what happens if one already exists, we should probably only create one if it doesn't exist

File: client/src/commands/testSmartContractCommand.ts [hunk 2]
```diff
@@ -276,8 +305,8 @@ async function installNodeModules(dir: string, language: string): Promise<void>
     let npmInstallOut: string;
 
     if (language === 'TypeScript') {
-        outputAdapter.log(LogType.INFO, 'Installing package dependencies including: fabric-network@1.4.0, fabric-client@1.4.0, @types/mocha');
-        npmInstallOut = await CommandUtil.sendCommandWithProgress('npm install && npm install --save-dev fabric-network@1.4.0 fabric-client@1.4.0 @types/mocha', dir, 'Installing npm and package dependencies in smart contract project');
+        outputAdapter.log(LogType.INFO, 'Installing package dependencies including: fabric-network@1.4.0, fabric-client@1.4.0, @types/mocha, ts-node, typescript');
+        npmInstallOut = await CommandUtil.sendCommandWithProgress('npm install && npm install --save-dev fabric-network@1.4.0 fabric-client@1.4.0 @types/mocha ts-node typescript', dir, 'Installing npm and package dependencies in smart contract project');
```
Comment (caroline-church) [inline client/src/commands/testSmartContractCommand.ts]: do we need the first npm install?

Comment (Jakeeyturner) [reply client/src/commands/testSmartContractCommand.ts]: First `npm install` has now been removed.

File: client/src/commands/testSmartContractCommand.ts (hunk 1 above)
Comment (Jakeeyturner) [reply client/src/commands/testSmartContractCommand.ts]: We now check if a `tsconfig.json` exists. If it already exists, we warn the user but continue and assume that the config will work.

**Pull request merged**